ICAFIX/hcp_fix_multi_run

```diff
@@ -498,317 +740,97 @@ for fmri in "${fmris[@]}" ; do
-## Run melodic on concatenated file
-## ---------------------------------------------------------------------------
+	# Delete some time series, resulting from the '--Oall' option in melodic, that aren't needed
+	# (these may only get created in the context of MIGP)
+	$FSLDIR/bin/imrm ${concatfmrihp}.ica/filtered_func_data.ica/alldat
+	$FSLDIR/bin/imrm ${concatfmrihp}.ica/filtered_func_data.ica/concat_data
 
-log_Debug_Msg "About to run melodic: Contents of ${concatfmrihp}.ica follow"
-if [ ! -z "${log_debugOn}" ] ; then
-	ls -lRa ${concatfmrihp}.ica
-fi
+	## ===========================================================================
+	## End skip in case of RunFIXOnly
+	## ===========================================================================
 
-#grab melodic from $PATH by default, don't hardcode it with respect to $FSLDIR
-#we need to do "if which ..." because the script currently uses ERR trap
-if which melodic &> /dev/null
-then
-	MELODIC=$(which melodic 2> /dev/null)
 else
-	#if it isn't even in $PATH, fall back on FSLDIR
-	MELODIC="${FSLDIR}/bin/melodic"
-fi
+	log_Msg "Skipping directly to FIX. All the files generated by ICA have to be present!"
 
-log_Msg "Running MELODIC located at: $MELODIC"
-log_Debug_Msg "Beginning of melodic version log, help, and checksum"
-if [ ! -z "${log_debugOn}" ] ; then
-	log_Debug_Msg "$MELODIC --version"
-	$MELODIC --version
-	log_Debug_Msg "$MELODIC --help"
-	$MELODIC --help
-	log_Debug_Msg "md5sum $MELODIC"
-	md5sum $MELODIC
-fi
-log_Debug_Msg "End of melodic version log, help, and checksum"
-
-# Use the Wishart-based dim estimate instead of the melodic-based estimate (--dim=${Dim})
-# Also turn off melodic VN and feed in the input that has already been VN'ed instead (--vn flag, which turns OFF melodic's vn)
-# Added "-m ${concatfmri}_brain_mask" to avoid memory issue - Takuya Hayahsi July 2018
-melodic_cmd=("${MELODIC}" -i "${concatfmrihp}_vnts" -o "${concatfmrihp}.ica/filtered_func_data.ica" --nobet --report --Oall --tr="${tr}" --vn --dim="${Dim}" -m "${concatfmri}_brain_mask")
-if [ ! -z "${log_debugOn}" ] ; then
-	melodic_cmd+=(--verbose --debug)
-fi
+	# Set the variables
+	ConcatFolder=`dirname ${ConcatName}`
+	ConcatNameNoExt=$($FSLDIR/bin/remove_ext $ConcatName)  # No extension, but still includes the directory path
+	concatfmri=`basename ${ConcatNameNoExt}`  # Directory path is now removed
+	concatfmrihp=${concatfmri}_hp${hp}
 
-#keep the existing handling code for melodic errors?
-debug_disable_trap
+	# Switch to concat folder
+	cd ${ConcatFolder}
 
-log_Msg "melodic_cmd: ${melodic_cmd[*]}"
-"${melodic_cmd[@]}"
-return_code=$?
-log_Msg "melodic has been run: return_code = ${return_code}"
-log_Debug_Msg "melodic has been run: Contents of ${concatfmrihp}.ica follow"
-if [ ! -z "${log_debugOn}" ] ; then
-	ls -lRa ${concatfmrihp}.ica
-fi
+	if [ -d "${concatfmrihp}.ica/fix" ]; then
+		mv "${concatfmrihp}.ica/fix" "${concatfmrihp}.ica/fix-bkp-$(date +"%Y-%m-%d.%H%M%S")"
```

Comment:
Do we want the pipeline to internally make a backup?  Granted, this option is a special case, but we generally don't like making backups the user doesn't know about.

Reply:
Backing up previous solution is important in the process of iteratively building and evaluating FIX models, which is the use case that this option is made for. We could add another optional parameter to give the user more control, but I suspect that whoever would use `--fix-only` would also want to have the previous fix solutions backed up. To make the backup explicit, I have now added a message in the log (it will be in the next PR):

```
        if [ -d "${concatfmrihp}.ica/fix" ]; then
		fix_backup_suffix="-bkp-$(date +"%Y-%m-%d.%H%M%S")"
		mv "${concatfmrihp}.ica/fix" "${concatfmrihp}.ica/fix${fix_backup_suffix}"
		log_Msg "previous fix solution found and renamed to fix${fix_backup_suffix}"
	fi
```

Reply:
Perhaps that behavior should be a separate option?

Reply:
@glasserm, we can make it a separate option, but I suspect it will be always used in the use cases for `--skip-ica`.

Reply:
Also, the option only makes sense when using `--skip-ica` as with a new ica the old fix does not refer to the same components any more.

Reply:
@glasserm Are we okay leaving the automatic backup stuff mandatory in this mode?  Log messages are not all that likely to get the user's attention unless the script fails.

If we are still fine with this (perhaps because of the limited use case of this feature), then I think it is ready for merge.

Reply:
I would prefer a separate option if that is okay.  I don't think one would always want to back up results.

Reply:
@demsarjure @sivcek Can you add a `--backup-fix-if-ica-skipped` boolean option that defaults to true?  Or, we could make a more general `--backup-existing-fix` option that defaults to false, and also works when not using `--skip-ica`.

Reply:
What happened to @coalsont's previous suggestion for an alternative name to `--skip-ica`?  I don't recall exactly what it was, but I do recall that he made a good argument for his alternative name.

Reply:
Or, to make it more scriptable, it could be something like `--fix-backup-suffix=<string>` or `--fix-backup-folder=<fix-bkp-something>`, and if it isn't null, names the backup folder accordingly, rather than generating it from `date`.  The second form could even take an absolute path, or be used to put all the backups into a `backups/` subfolder instead of littering the `${concatfmrihp}.ica` folder directly.

Reply:
Ok, found it.  It was `--reuse-existing-ica`:
https://github.com/Washington-University/HCPpipelines/pull/297#discussion_r1723691813

Reply:
@mharms I had resolved it as somewhat minor and we seemed close to merging, but I have unresolved it again since you seem to have an opinion on it.